### DISCUSSION TITLE
Can CrySPY perform variable component prediction or variable Z-value prediction？

### DISCUSSION THREAD

**Discussion by Lamed-git** [Tomoki-YAMASHITA/CrySPY#62] [General]
After consulting the manuals and examples, I found that CrySPY can only be given a fixed natot and nat for structural prediction. I wonder if variable components or variable Z value prediction can be added.

**Comment by Tomoki-YAMASHITA** [comment]
Variable Z means variable number of formula units?
CrySPY 1.2.5, which was just released a couple of days ago, can treat variable composition EA. (not random search)
However, only binary systems are available for use, and we haven't check the bugs carefully.

See,
https://github.com/Tomoki-YAMASHITA/CrySPY/discussions/60#discussioncomment-9339083 

We are working on adapting it to handle multi-component systems.

**Reply by Lamed-git** [reply]
Thank you for your reply. You are right, variable Z means variable number of formula units (refine the chemical composition and change the total atomic number within a specific range). 
After your guidance, I know how to use CrySPY to make variable component prediction, looking forward to the expansion of multi-component systems.
